src/util/ChartUtils.ts

```diff
@@ -1251,11 +1252,7 @@ export const parseSpecifiedDomain = (specifiedDomain: any, dataDomain: any, allo
  * @param  {Boolean} isBar if items in axis are bars
  * @return {Number} Size
  */
-export const getBandSizeOfAxis = (
-  axis?: BaseAxisProps,
-  ticks?: Array<TickItem>,
-  isBar?: boolean,
-): number | undefined => {
+export const getBandSizeOfAxis = (axis?: XAxisProps, ticks?: Array<TickItem>, isBar?: boolean): number | undefined => {
```

Comment:
now the type isn't accurate because this function isn't called for only `XAxis`

We need a union type + type narrowing

So
```ts
// defined above or elsewhere
type CartesianAxisProps = XAxisProps | YAxisProps | ZAxisProps


// in function
 if (axis && 'width' in axis && axis.width && ticks) {
```

Reply:
I applied the fixes. Can you verify if it is correct now? Pls